src/constraint/spec.ts

```diff
@@ -429,37 +430,64 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
   {
     name: 'omitNonPositionalOrFacetOverPositionalChannels',
     description: 'Do not use non-positional channels unless all positional channels are used',
-    properties: [Property.CHANNEL],
+    properties: [Property.CHANNEL, Property.MARK],
     allowWildcardForProperties: false,
     strict: false,
     satisfy: (specM: SpecQueryModel, _: Schema, opt: QueryConfig) => {
-      const encodings = specM.specQuery.encodings;
-      let hasNonPositionalChannelOrFacet = false;
-      let hasEnumeratedNonPositionOrFacetChannel = false;
-      let hasX = false, hasY = false;
-      for (let i = 0; i < encodings.length; i++) {
-        const encQ = encodings[i];
-        if (isValueQuery(encQ) || (isDisabledAutoCountQuery(encQ))) {
-          continue; // ignore skipped encoding
-        }
+      const mark = specM.getMark();
 
-        const channel = encQ.channel;
-        if (channel === Channel.X) {
-          hasX = true;
-        } else if (channel === Channel.Y) {
-          hasY = true;
-        } else if (!isWildcard(channel)) {
-          // All non positional channel / Facet
-          hasNonPositionalChannelOrFacet = true;
-          if (specM.wildcardIndex.hasEncodingProperty(i, Property.CHANNEL)) {
-            hasEnumeratedNonPositionOrFacetChannel = true;
+      switch(mark) {
+        case Mark.RECT:
+          // RECT is a special exception, as it can be used in a 1xD manner
```

Comment:
I'm not sure if this constraint should be relaxed for rect?

If there is rect 1D with color, it is still better to use bar 2D (move color to another position channel) anyway?

Reply:
If I recall correctly, in one of our previous conversations you mentioned this 1D w/ color should be a possibility (and that's why I relaxed the constraint). I think it does make more sense to just use a bar in that case, however.

Reply:
>  I think it does make more sense to just use a bar in that case, however.

I see.  I agree that bar still makes more sense anyway. 
Since this is a non-strict constraint, maybe we don't have to create a special case for rect?


In a way, point with x and size/color is a fair game too.  It's just that scatterplot is almost always more reasonable.  So it seems a bit inconsistent to relax this constraint only for rect.

Reply:
That sounds fair, I will remove this case